lib/runAll.js

```diff
@@ -107,14 +113,40 @@ export const runAll = async (
     return ctx
   }
 
-  const stagedFileChunks = chunkFiles({ baseDir: gitDir, files, maxArgLength, relative })
-  const chunkCount = stagedFileChunks.length
-  if (chunkCount > 1) debugLog(`Chunked staged files into ${chunkCount} part`, chunkCount)
+  const configDir2config = {
+    [gitDir]: config,
+  }
+  const configDir2Files = {
+    [gitDir]: [],
+  }
+  if (sparseConfig) {
+    for (const file of files) {
+      const absoluteFilepath = path.resolve(gitDir, file)
+      const subConfigInfo = await loadConfig({
+        searchStartPath: path.dirname(absoluteFilepath),
+        logger,
+      })
```

Comment:
I wonder if we could do this always, also in the case of non-sparse mode. So if we move all the config loading here so that the default "root" config is always loaded, and then in the case of sparse mode more files are discovered.

I hope you understand what I mean.

Reply:
Regarding this, how does supplying both a "root" config file as well as multiple nested ones using `--sparse-config` work?

I would assume the "root" config would be applied to all files unless there's a deeper config file, which would then be used instead.

How about supplying a config directly via the JS API with `lintStaged({ config: { ... } })`?

Reply:
Also, I think all configuration files should be validated.

Reply:
> the default "root" config is always loaded

This is a pretty trick question, let me answer it in another comment.

> How about supplying a config directly via the JS API with lintStaged({ config: { ... } })?

If cloest configuration file is found, use the configuration. If no configuration can be loaded, `config` passed will be used as a fallback

> Also, I think all configuration files should be validated.

Yes, let's do this.